privacy/scrubber.go

```diff
@@ -97,23 +106,37 @@ func (scrubber) ScrubRequest(bidRequest *openrtb2.BidRequest, enforcement Enforc
 		}
 		if bidRequest.User != nil {
 			bidRequest.User.Data = nil
-			bidRequest.User.Ext = scrubExtIDs(bidRequest.User.Ext, "data")
 			bidRequest.User.ID = ""
 			bidRequest.User.BuyerUID = ""
 			bidRequest.User.Yob = 0
 			bidRequest.User.Gender = ""
 			bidRequest.User.Keywords = ""
 			bidRequest.User.KwArray = nil
+
+			_, hasField := userExtParsed["data"]
+			if hasField {
+				delete(userExtParsed, "data")
+				userExtModified = true
+			}
 		}
 	}
 	if enforcement.Eids {
 		//transmitEids covers user.eids and user.ext.eids
 		if bidRequest.User != nil {
 			bidRequest.User.EIDs = nil
-			bidRequest.User.Ext = scrubExtIDs(bidRequest.User.Ext, "eids")
+			_, hasField := userExtParsed["eids"]
+			if hasField {
+				delete(userExtParsed, "eids")
+				userExtModified = true
+			}
 		}
 	}
 
+	if userExtModified {
+		userExt, _ := json.Marshal(userExtParsed)
+		bidRequest.User.Ext = userExt
```

Comment:
The bidRequest is absolutely at least a shallow copy at this point. I'm good with adding a shallow clone here to be certain if it's not otherwise proven. We'll replace it in the near future with the RequestWrapper and the cost of cloning the User object doesn't seem too concerning.

Reply:
Added clones.